### PR TITLE
Start adding special features examples

### PR DESCRIPTION
**Pull request by aguspe**
### Description
Right now none of the example features under Chrome have examples for users to follow, this PR adds the casting example

### Motivation and Context
The goal of this PR is to provide users with an overview of what special features they can use with Chrome

Screenshot 2024-05-08 at 16 43 42

### Types of changes

- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

___

### **PR Type**
enhancement, documentation

___

### **Description**
- Added a new Ruby test case for Chrome casting functionality, demonstrating device discovery and tab mirroring.
- Updated English, Japanese, Portuguese, and Chinese Chrome documentation to reference the new Ruby casting example.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>chrome_spec.rb</strong><dd><code>Add Ruby example for casting in Chrome</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/ruby/spec/browsers/chrome_spec.rb

<li>Added a new test case under 'Special Features' for casting <br>functionality in Chrome.<br> <li> The test initializes a Chrome driver, retrieves cast sinks, and <br>performs tab mirroring if devices are available.<br> <li> Ensures no exceptions are raised during the start and stop of casting.<br> <br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1711/files#diff-f3e68237826e1b64c37885dc12602e936e6e4ea29da653aeb659d466a465ffca">+13/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>chrome.en.md</strong><dd><code>Update English Chrome documentation with new Ruby example</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/browsers/chrome.en.md

<li>Updated the Ruby code block reference to point to the new casting <br>example.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1711/files#diff-eccc7d2ca21e26be42c8c903fa062012754d5ea90f001a432f4f9961e87b078c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>chrome.ja.md</strong><dd><code>Update Japanese Chrome documentation with new Ruby example</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/browsers/chrome.ja.md

<li>Updated the Ruby code block reference to point to the new casting <br>example.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1711/files#diff-85cef9cb445defdf9de42ecc0dbd533e2ef2b0b47bcd8fb2aecc09f2114b3238">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>chrome.pt-br.md</strong><dd><code>Update Portuguese Chrome documentation with new Ruby example</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/browsers/chrome.pt-br.md

<li>Updated the Ruby code block reference to point to the new casting <br>example.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1711/files#diff-4d648f7aea25c7e23a591cdb153ae6a6857d909ab673354eafdf3c5a3b4c4aa7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>chrome.zh-cn.md</strong><dd><code>Update Chinese Chrome documentation with new Ruby example</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/browsers/chrome.zh-cn.md

<li>Updated the Ruby code block reference to point to the new casting <br>example.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1711/files#diff-1aff253f48970f96fb108fece2476c5e0d815aec68c4275a2746b2bad32a377f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by harsha509 — APPROVED** [review]
Thank you @aguspe !

**Review by harsha509 — COMMENTED** [review]
HI @aguspe,

Can we just include code from L119-L122 Or L124?

**Comment by aguspe** [comment]
> HI @aguspe,
> 
> Can we just include code from L119-L122 Or L124?

Hi @harsha509 thank you so much for the question, I think the best is still L119-L124 because in the current implementation if people try just to use @driver.cast_sinks it will actually not work

I made a question regarding this on the slack channel to see how much we needed to expand the example for the users, here is the message: https://seleniumhq.slack.com/archives/C0K0BK8MQ/p1715280896233389

**Comment by harsha509** [comment]
> > HI @aguspe,
> > Can we just include code from L119-L122 Or L124?
> 
> Hi @harsha509 thank you so much for the question, I think the best is still L119-L124 because in the current implementation if people try just to use @driver.cast_sinks it will actually not work
> 
> I made a question regarding this on the slack channel to see how much we needed to expand the example for the users, here is the message: https://seleniumhq.slack.com/archives/C0K0BK8MQ/p1715280896233389

L119-L124 looks fine to me...

**Comment by aguspe** [comment]
> > > HI @aguspe,
> > > Can we just include code from L119-L122 Or L124?
> > 
> > 
> > Hi @harsha509 thank you so much for the question, I think the best is still L119-L124 because in the current implementation if people try just to use @driver.cast_sinks it will actually not work
> > I made a question regarding this on the slack channel to see how much we needed to expand the example for the users, here is the message: https://seleniumhq.slack.com/archives/C0K0BK8MQ/p1715280896233389
> 
> L119-L124 looks fine to me...

Perfect thank you so much @harsha509, let me know if there are other questions

File: website_and_docs/content/documentation/webdriver/browsers/chrome.en.md [hunk 1]
```diff
@@ -379,7 +379,7 @@ You can drive Chrome Cast devices, including sharing tabs
 {{< badge-code >}}
 {{< /tab >}}
 {{< tab header="Ruby" >}}
-{{< badge-code >}}
+{{< gh-codeblock path="examples/ruby/spec/drivers/chrome_spec.rb#L118-L125" >}}
```
Comment (harsha509) [inline website_and_docs/content/documentation/webdriver/browsers/chrome.en.md]: So can you change these ?

**Review by harsha509 — CHANGES_REQUESTED** [review]

File: website_and_docs/content/documentation/webdriver/browsers/chrome.en.md (hunk 1 above)
Comment (aguspe) [reply website_and_docs/content/documentation/webdriver/browsers/chrome.en.md]: Hi @harsha509 is updated now, have a great weekend!

**Review by harsha509 — APPROVED** [review]

**Pull request merged**